api/types.go

```diff
@@ -371,6 +371,7 @@ type ProgressResponse struct {
 	Digest    string `json:"digest,omitempty"`
 	Total     int64  `json:"total,omitempty"`
 	Completed int64  `json:"completed,omitempty"`
+	Type 	  string `json:"type,omitempty"`
```

Comment:
Is there a way we can avoid this? e.g. perhaps look to see if we set `digest` which might mean it's a pull?